homeassistant/components/rainmachine/sensor.py

```diff
@@ -317,6 +377,14 @@ def update_from_latest_data(self) -> None:
             self._attr_native_value = self.coordinator.data.get("system", {}).get(
                 "flowSensorWateringClicks"
             )
+        elif self.entity_description.key == TYPE_LAST_LEAK_DETECTED:
+            self._attr_native_value = utc_from_timestamp(
+                self.coordinator.data.get("system", {}).get("lastLeakDetected")
```

Comment:
Won't `utc_from_timestamp` blow up if this ends up being `None`?

Reply:
Yep. Resolved in the restuctured method. See comment below.